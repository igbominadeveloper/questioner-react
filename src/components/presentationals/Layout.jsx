import React, { Fragment } from 'react';

import NavBar from './Navbar/Navbar';
import Footer from './Footer/Footer';

const Layout = ({ children }) => {
  return (
    <Fragment>
      <div className="container-fluid" data-test="container-fluid">
        <NavBar data-test="navbar" />
        {children}
      </div>
      <Footer />
    </Fragment>
  );
};

export default Layout;
