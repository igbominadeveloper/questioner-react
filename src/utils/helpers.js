import jwt from 'jsonwebtoken';

export const decodeToken = token => {
  return jwt.decode(token);
};

export const setItem = (name, value) => {
  localStorage.setItem(name, value);
};

export const getItem = name => {
  const item = localStorage.getItem(name);
  return item;
};

export const destroyItem = item => {
  return localStorage.removeItem(item);
};

export const clearLocalStorage = () => {
  return localStorage.clear();
};
