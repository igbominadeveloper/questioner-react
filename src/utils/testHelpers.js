import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const mockStore = configureStore([thunk]);

export const setupStore = initialState => mockStore(initialState);

export const findByTestAttribute = (component, attribute) => {
  const wrapper = component.find(`[data-test='${attribute}']`);
  return wrapper;
};
