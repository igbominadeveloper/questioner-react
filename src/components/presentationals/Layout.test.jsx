import React from 'react';
import { shallow } from 'enzyme';

import Layout from './Layout';

import { findByTestAttribute } from '../../utils/testHelpers';

const component = shallow(<Layout />);

describe('Custom Search', () => {
  it('renders without errors', () => {
    expect(component).toMatchSnapshot();
  });

  it('renders a nav html element', () => {
    const wrapper = findByTestAttribute(component, 'container-fluid');
    expect(wrapper.length).toBe(1);
  });

  it('renders the navbar', () => {
    const wrapper = findByTestAttribute(component, 'navbar');
    expect(wrapper.length).toBe(1);
  });
});
