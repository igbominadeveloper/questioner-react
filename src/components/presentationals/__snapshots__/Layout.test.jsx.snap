// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Search renders without errors 1`] = `
<Fragment>
  <div
    className="container-fluid"
    data-test="container-fluid"
  >
    <Navbar
      data-test="navbar"
      onClick={[Function]}
    />
  </div>
  <Footer />
</Fragment>
`;
